src/manifest.js

```diff
@@ -1,5 +1,6 @@
 let cachedRemoteManifest;
```

Comment:
Could we simplify this by not creating this `cachedRemoteManifest` and instead always cache the `fetchManifestPromise` (don't reset it to null below). Then always return the one promise.